stdlib/public/runtime/Heap.cpp

```diff
@@ -23,12 +23,11 @@
 using namespace swift;
 
 void *swift::swift_slowAlloc(size_t size, size_t alignMask) {
-  // FIXME: use posix_memalign if alignMask is larger than the system guarantee.
-  void *p = malloc(size);
+  void *p = AlignedAlloc(size, alignMask + 1);
```

Comment:
Is `alignMask` guaranteed to be a power-of-two-minus-one?

Reply:
That should be fine. What might be a problem is that `posix_memalign` requires that the alignment be no smaller than `sizeof(void*)`. We have at least one place that calls `swift_slowAlloc(size, 0)` because it needs no alignment constraints.

Reply:
`AlignedAlloc` makes sure the alignment is sufficiently large before calling through to `posix_memalign`.

Reply:
That is unfortunate for performance. It would be better if we could satisfy posix_memalign's constraints directly. That would be a much wider change.